hibernate-reactive-core/src/test/java/org/hibernate/reactive/OrderTest.java

```diff
@@ -0,0 +1,242 @@
+						session -> session.createSelectionQuery( "select title from Book", String.class )
+								.getResultList()
+								.invoke( list -> assertTrue( list.size() == 2 ) )
+								.chain( vv -> {
+									MappingMetamodelImpl metamodel = (MappingMetamodelImpl) getSessionFactory().getMetamodel();
+									EntityDomainType<Book> bookType = metamodel.getJpaMetamodel()
+											.findEntityType( Book.class );
+									SingularAttribute<? super Book, ?> title = bookType.findSingularAttribute( "title" );
+									return session
+											.createSelectionQuery( "from Book", Book.class )
+											.setOrder( Order.asc( title ) )
+											.getResultList();
+								} ).invoke( books -> {
+									assertEquals( "Hibernate in Action", books.get( 0 ).title );
+									assertEquals( "Java Persistence with Hibernate", books.get( 1 ).title );
+								} )
+								.chain( vv -> {
+									MappingMetamodelImpl metamodel = (MappingMetamodelImpl) getSessionFactory().getMetamodel();
+									EntityDomainType<Book> bookType = metamodel.getJpaMetamodel()
+											.findEntityType( Book.class );
+									SingularAttribute<? super Book, ?> title = bookType.findSingularAttribute( "title" );
+									return session
+											.createSelectionQuery( "from Book", Book.class )
+											.setOrder( Order.desc( title ) )
+											.getResultList();
+								} ).invoke( books -> {
+									assertEquals( "Hibernate in Action", books.get( 1 ).title );
+									assertEquals( "Java Persistence with Hibernate", books.get( 0 ).title );
+								} )
+								.chain( vv -> {
+									MappingMetamodelImpl metamodel = (MappingMetamodelImpl) getSessionFactory().getMetamodel();
+									EntityDomainType<Book> bookType = metamodel.getJpaMetamodel()
+											.findEntityType( Book.class );
+									SingularAttribute<? super Book, ?> isbn = bookType.findSingularAttribute( "isbn" );
+									return session
+											.createSelectionQuery( "from Book", Book.class )
+											.setOrder( Order.asc( isbn ) )
+											.getResultList();
+								} ).invoke( books -> {
+									assertEquals( "Hibernate in Action", books.get( 1 ).title );
+									assertEquals( "Java Persistence with Hibernate", books.get( 0 ).title );
+								} )
+								.chain( vv -> {
+									MappingMetamodelImpl metamodel = (MappingMetamodelImpl) getSessionFactory().getMetamodel();
+									EntityDomainType<Book> bookType = metamodel.getJpaMetamodel()
+											.findEntityType( Book.class );
+									SingularAttribute<? super Book, ?> isbn = bookType.findSingularAttribute( "isbn" );
+									return session
+											.createSelectionQuery( "from Book", Book.class )
+											.setOrder( Order.desc( isbn ) )
+											.getResultList();
+								} ).invoke( books -> {
+									assertEquals( "Hibernate in Action", books.get( 0 ).title );
+									assertEquals( "Java Persistence with Hibernate", books.get( 1 ).title );
+								} )
+				) )
```

Comment:
If you create a single test method for each block, the whole class becomes much more readable and easier to understand (same is true for the other tests).